File: 序号/1813. 句子相似性 III.py
class Solution:
    def areSentencesSimilar(self, sentence1: str, sentence2: str) -> bool:
        words1, words2 = sentence1.split(" "), sentence2.split(" ")
        if len(words1)>len(words2):
            words1,words2=words2,words1
        for i in range(len(words1)+1):  # words1 前i个单词在words2的首部，其余在尾部
            f = True
            for j in range(len(words1)):
                if j < i:
                    if words1[j] == words2[j]:
                        continue
                    else:
                        f = False
                        break
                else:
                    if words1[j - len(words1)] == words2[j - len(words1)]:
                        continue
                    else:
                        f = False
                        break
            if f:
                return True
        return False

File: 序号/1813. 句子相似性 III.cpp
class Solution {
 private:
  vector<string> split(string& s, char dlm) {
    stringstream ss(s);
    string buffer;
    vector<string> v;
    while (getline(ss, buffer, dlm)) {
      v.emplace_back(buffer);
    }
    return v;
  }

 public:
  bool areSentencesSimilar(string sentence1, string sentence2) {
    vector<string> v1 = split(sentence1, ' '), v2 = split(sentence2, ' ');
    int i = 0, j = 0;
    while (i < v1.size() && i < v2.size() && v1[i] == v2[i]) i++;
    while (v1.size() - j -1>= 0 && v2.size() - j-1 >= 0 && v1[v1.size() - 1 - j] == v2[v2.size() - 1 - j]) {
        j++;
        cout<<j<<','<<v1.size()<<','<<v2.size()<<','<<(v1.size() - j -1)<<','<<( v2.size() - j-1 )<<endl;
    }
    return (i + j) >= min(v1.size(), v2.size());
  }
};